redbot/setup.py

```diff
@@ -248,11 +248,17 @@ async def remove_instance(
     remove_datapath: Optional[bool] = None,
 ):
     data_manager.load_basic_configuration(instance)
+    backend = get_current_backend(instance)
 
     if interactive is True and delete_data is None:
-        delete_data = click.confirm(
-            "Would you like to delete this instance's data?", default=False
-        )
+        if backend != BackendType.JSON:
+            delete_data = click.confirm(
+                "Would you like to delete this instance's data? The database server must be running for this to work. If you do not have a database server, select No.", default=False
```

Comment:
There is no need to mention not having the database server, it should be clear enough with just the first part and otherwise, it is also quite long.
```suggestion
                "Would you like to delete this instance's data?"
                " The database server must be running for this to work.",
                default=False,
```
Also, this code suggestion fixes issues with the black style.